diff --git a/src/main.cpp b/src/main.cpp
index b811ed5..668f449 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -6,6 +6,8 @@
 #include <boost/algorithm/string/replace.hpp>
 #include <boost/filesystem.hpp>
 #include <boost/filesystem/fstream.hpp>
+#include <boost/bind.hpp>
+using namespace boost::placeholders;
 
 #include "chainparams.h"
 #include "checkpoints.h"
diff --git a/src/rpcclient.cpp b/src/rpcclient.cpp
index a383e6a..fef3d98 100644
--- a/src/rpcclient.cpp
+++ b/src/rpcclient.cpp
@@ -35,9 +35,9 @@ Object CallRPC(const string& strMethod, const Array& params)
 
     // Connect to localhost
     bool fUseSSL = GetBoolArg("-rpcssl", false);
-    asio::io_service io_service;
-    ssl::context context(io_service, ssl::context::sslv23);
+    ssl::context context(ssl::context::sslv23);
     context.set_options(ssl::context::no_sslv2);
+    asio::io_service io_service;
     asio::ssl::stream<asio::ip::tcp::socket> sslStream(io_service, context);
     SSLIOStreamDevice<asio::ip::tcp> d(sslStream, fUseSSL);
     iostreams::stream< SSLIOStreamDevice<asio::ip::tcp> > stream(d);
